bigchaindb/util.py

```diff
@@ -595,38 +143,27 @@ def verify_vote_signature(block, signed_vote):
     vk_base58 = signed_vote['node_pubkey']
 
     # immediately return False if the voter is not in the block voter list
-    if vk_base58 not in block['block']['voters']:
+    if vk_base58 not in voters:
         return False
 
     public_key = crypto.VerifyingKey(vk_base58)
     return public_key.verify(serialize(signed_vote['vote']), signature)
 
 
-def transform_create(tx):
-    """Change the owner and signature for a ``CREATE`` transaction created by a node"""
-
-    # XXX: the next instruction opens a new connection to the DB, consider using a singleton or a global
-    #      if you need a Bigchain instance.
-    b = bigchaindb.Bigchain()
-    transaction = tx['transaction']
-    payload = None
-    if transaction['data'] and 'payload' in transaction['data']:
-        payload = transaction['data']['payload']
-    new_tx = create_tx(b.me, transaction['fulfillments'][0]['owners_before'], None, 'CREATE', payload=payload)
-    return new_tx
-
-
 def is_genesis_block(block):
```

Comment:
Not very proud of this method.
It handles both a dict-block as well as a Block-block.

Ideally, it should only handle Block-block.